### PR TITLE
Add DApps board

### LINKED ISSUE #1168: DApp board
### Is your feature request related to a problem?

_No response_

### Describe the solution you'd like

Add a feature to list available Dapps
Reuse Dapp definition data transaction (ex mainnet: 0000679EF0AF4ED6EDB1790B8E082880BD18632BB66190DE93422C8CFC7E57D678EE) and add properties:
- Name of the DApp
- Category
- Description
- Url icon

Add a webview to list DApps / Select DApps / Manage Favorites

### Additional context

_No response_

### Epic

_No response_

### PR DESCRIPTION
**Pull request by redDwarf03**
# Description

Fixes #1168 

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by redDwarf03** [comment]
Based on bridge integration branch
- [x] Issue with loader (to be fixed in bridge integration branch)
- [x] Manage the dapp launch in desktop mode

**Comment by redDwarf03** [comment]
from a UX point of view, it's not a good idea to use a full `CupertinoModalBottomSheet`, because when we scroll the dapp, it closes the modal.

**Comment by redDwarf03** [comment]
Dpix app is not compliant with mobile platform... don't integrate before app updates

**Comment by redDwarf03** [comment]
> Dpix app is not compliant with mobile platform... don't integrate before app updates
Use feature flag

**Review by Chralu — APPROVED** [review]

File: lib/ui/views/sheets/dapp_sheet.dart [hunk 1]
```diff
@@ -90,30 +99,31 @@ class DAppSheetState extends ConsumerState<DAppSheet>
...
-      final bridgeFlag = await ref.watch(
-        getFeatureFlagProvider(
-          networkSettings.network,
-          applicationCode,
-          featureCode,
-        ).future,
-      );
-      if (bridgeFlag == null || bridgeFlag == false || !DAppSheet.isAvailable) {
-        dappKey = '${dappKey}Ext';
+      if (widget.featureCode != null) {
```
Comment (Chralu) [inline lib/ui/views/sheets/dapp_sheet.dart]: @redDwarf03 I don't  get it...

Comment (redDwarf03) [reply lib/ui/views/sheets/dapp_sheet.dart]: The suffix `Ext` helps to find the url to launch (via launch button) if dapp is not managed directly in aeWallet.
See: https://testnet.archethic.net/explorer/transaction/000087F7890DB91FF666851B3D4BF618BBC4771218BA6A319AE3EF55AFA1718C2EA1

**Review by Chralu — DISMISSED** [review]
`dapp_sheet.dart` needs a bit of cleanup : I think special cases dedicated to Bridge should be extracted.

**Comment by redDwarf03** [comment]
> `dapp_sheet.dart` needs a bit of cleanup : I think special cases dedicated to Bridge should be extracted.

Done. please review it

**Pull request merged**